Dialog/GraphicManagerDialog.cpp

```diff
@@ -554,25 +554,27 @@ void GraphicManagerDialog::DeltmpEntryTile(int tileId)
             }
             QByteArray data;
             int old_tilenum = tmpEntry.TileDataSizeInByte / 32;
-            data = tmpEntry.tileData.mid(0, 32 * (tileId - startid)) +
-                    tmpEntry.tileData.mid(32 * (tileId + 1 - startid), 32 * (old_tilenum + startid - tileId - 1));
+            data = tmpEntry.tileData.left(32 * (tileId - startid)) + tmpEntry.tileData.right(32 * (old_tilenum + startid - tileId - 1));
             tmpEntry.tileData = data;
 
             tmpEntry.TileDataRAMOffsetNum += 1;
             tmpEntry.TileDataSizeInByte -= 32;
 
             // update mapping data
-            int width = tmpEntry.optionalGraphicWidth;
-            for (int h = 0; h < tmpEntry.optionalGraphicHeight; h++)
+            for (int w = 0; w < tmpEntry.mappingData.size(); w++)
             {
-                for (int w = 0; w < width; w++)
+                unsigned short data = tmpEntry.mappingData[w];
+                unsigned short id = data & 0x3FF;
+                if (id < tileId)
                 {
-                    unsigned short data = tmpEntry.mappingData[w + h * width];
-                    int id = data & 0x3FF;
-                    if ((tmpEntry.mappingData[w + h * width] & 0x3FF) <= tileId)
-                    {
-                        tmpEntry.mappingData[w + h * width] = (data & 0xFC00) | ((id + 1) & 0x3FF);
-                    }
+                    tmpEntry.mappingData[w] = (data & 0xFC00) | ((id + 1) & 0x3FF);
+                }
+                else if (id == tileId)
+                {
+                    // something went wrong in the previous code if this part of code gets executed
```

Comment:
Can we print a popup with a warning if the code reached this point then? It is better if we can recognize some case later when this happens, and update the functionality if necessary

Reply:
done